### DISCUSSION TITLE
How to strafe/move dynamic body horizontally

### DISCUSSION THREAD

**Discussion by rossmarinaro** [enable3d/enable3d#264] [Help / Questions]
Following examples of first, and third person camera, I've found myself stumped on how to achieve a sidestep/strafe along a horizontal axis.  Moving forward works, and to go back, just inverse the calculation, however I'm quite confused on how to rotate the body to now update its axis properly, and even attempted to remove and add a physics body, and change collision flags. below is the code to move the player, but again, left and right don't properly work:

 movePlayer (forceX, forceY)
{
 const raycaster = new THREE.Raycaster();
 const direction = this.third.camera.getWorldDirection(raycaster.ray.direction),
          x = direction.x * 100, 
          y = player.position.y, 
          z = direction.z * 100;

    if (forceX > 40) //right
     player.body.setVelocityX(-100);  //not consistent
    else if (forceX < -40) //left
      player.body.setVelocityX(100); //not consistent
    else if (forceY < -40) //down
      player.body.setVelocity(x, y, z); //works
    else if (forceY > 40) //up
      player.body.setVelocity(-x, y, -z); //works
}

**Comment by rossmarinaro** [comment]
was wondering if anyone had any input on this, I know its not exactly strictly enable3d, but haven't really grapsed a workable solution, although I have a feeling the "setRotationFromEuler" may be what i'm looking for, along with setAngularVelocityY. The body doesn't seem to update its rotation.

**Reply by luddensdesir** [reply]
I think your problem might be that you're not setting the right collision flag.

https://enable3d.io/docs.html#body-types

I think you're looking for 2. So this is how I initialize my object: 


    const box = this.physics.add.box(
      { x: 0, y: 2, z: -5, width: 1, height: 1, depth: 3, mass :1, collisionFlags: 0 },
      { lambert: { color: 'red', transparent: true, opacity: 0.5 } }
    );

And I have a special function for setting the mass:
 
```
const setMass = (body, mass) => {
  body['mass'] = mass;
  body.ammo.setMassProps(mass, new threeVector(0,0,0));
  body.ammo.getCollisionShape().calculateLocalInertia( mass, new threeVector(0,0,0))
};
```

**Comment by rossmarinaro** [comment]
What does mass have to do with rotating the body? setting collision flags back and forth between 0, and 2 isn't a solution, the body does not change rotation, and the player now falls through space. If I set the cam to kinematic, I now have the problem of the player now not colliding with walls, another issue entirely. I'm simply looking to move in all directions, consistently. forward and back work fine, but horizontally is not updating the body and always goes the same direction regardless of camera direction.
